lib/components/dashboard/degree-info/DistributionBarsJSX.tsx

```diff
@@ -237,13 +237,19 @@ const DistributionBarsJSX: FC<{ major: Major }> = ({ major }) => {
     req: requirements,
     reqGroup: [string, requirements[]],
   ) => {
-    if (req.pathing) {
-      let [requirement, ...focus_areas] = reqGroup[1];
-      for (let focus_area of focus_areas) {
-        if (focus_area.fulfilled_credits >= focus_area.required_credits) {
-          reqGroup[1] = [requirement, focus_area];
+    if (req.pathing !== undefined) {
+      let numPaths = req.pathing;
+      let [requirement, ...paths] = reqGroup[1];
+      let satisfiedFineRequirements = [requirement];
+      for (let path of paths) {
+        if (path.fulfilled_credits >= path.required_credits) {
+          satisfiedFineRequirements.push(path);
+          numPaths -= 1;
```

Comment:
Could numPaths be negative?

Reply:
Also what is the purpose of decreasing numPaths here?

Reply:
If one any path gets satisfied, that is one less path left to satisfy. If numPaths reduces to 0, that means they user has satisfied everything they need to satisfy. Since that for loop runs every time a course is added, if we specify a positive value from the beginning, numPaths can never be negative.

Reply:
Ohhhh ok. What is a situation that this covers? I think you said this in a meeting, but I'd like a refresher, ty!

Reply:
Yeah this is basically so that we don't need to include every single combination of focus areas in the fine requirements. In this case, if the pathing value is 2, the first 2 fine requirements that are satisfied will be considered and marked so. Then the distribution will be marked as complete only if the number of fine requirements equal to the pathing number are satisfied.

Reply:
Hmmm, ok I see. I think we should still do something about the situations where we have to create Distributions just for pathing. Can we find a way to combine pathing and non-pathing distributions? If not we can go with this for now ^.^